pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1442,22 +1443,36 @@ func getCDFromClient(c client.Client) *hivev1.ClusterDeployment {
 	return nil
 }
 
+func createGlobalPullSecretObj(secretType corev1.SecretType, name, key, value string) *corev1.Secret {
+	secret := &corev1.Secret{
+		Type: secretType,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: constants.HiveNamespace,
+		},
+		Data: map[string][]byte{
+			key: []byte(value),
+		},
+	}
+	return secret
+}
+
 func TestMergePullSecrets(t *testing.T) {
 
 	tests := []struct {
 		name             string
 		localPullSecret  string
 		globalPullSecret string
 		mergedPullSecret string
-		existingCD       []runtime.Object
+		existingObjs     []runtime.Object
 		expectedErr      bool
 		expectedMismatch bool
 	}{
 		{
 			name:             "merged pull secret should be be equal to local secret",
 			localPullSecret:  `{"auths": {"registry.svc.ci.okd.org": {"auth": "dXNljlfjldsfSDD"}}}`,
 			mergedPullSecret: `{"auths": {"registry.svc.ci.okd.org": {"auth": "dXNljlfjldsfSDD"}}}`,
-			existingCD: []runtime.Object{
+			existingObjs: []runtime.Object{
 				func() *hivev1.ClusterDeployment {
 					cd := getCDWithoutPullSecret()
 					cd.Spec.PullSecret = &corev1.LocalObjectReference{
```

Comment:
Why are the tests explicitly getting a clusterdeployment without a pull secret and then adding the pull secret rather than using `testClusterDeployment`, which already has the pull secret added?

Reply:
We are doing some testing without local pull secret too, hence we need `getCDWithoutPullSecret`

Reply:
I understand the use for the `getCDWithoutPullSecret` function. However, here, you ultimately build a clusterdeployment **with** a pull secret. There is no need to create a clusterdeployment explicitly without a pull secret and then add the pull secret rather rather than using the function that already exists that creates a clusterdeployment with the pull secret.
```
existingObjs: []runtime.Object{
  testClusterDeployment(),
},
```